### PR TITLE
🌱 Bump golang.org/x/net to v0.23.0 and golang to v1.21.9

### PR DESCRIPTION
**Pull request by kashifest**
Fixes # CVE-2023-45288

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -77,7 +77,7 @@ require (
 	go.opentelemetry.io/otel/trace v1.20.0 // indirect
 	go.opentelemetry.io/proto/otlp v1.0.0 // indirect
 	golang.org/x/exp v0.0.0-20230905200255-921286631fa9 // indirect
-	golang.org/x/net v0.22.0 // indirect
+	golang.org/x/net v0.23.0 // indirect
```
Comment (tuminoid) [inline go.mod:80]: Please bump all go.mods, x/net is present in other go.mods files too.

Comment (kashifest) [reply go.mod:80]: done

**Review by tuminoid — COMMENTED** [review]
/lgtm

**Review by lentzi90 — COMMENTED** [review]
/approve

**Pull request merged**